Sources/Conformance/test_messages_proto3.pb.swift

```diff
@@ -1673,480 +1150,987 @@ struct ProtobufTestMessages_Proto3_TestAllTypes: SwiftProtobuf.Proto3Message, Sw
       }
-      if storage._fieldName10 != 0 {
-        try visitor.visitSingularInt32Field(value: storage._fieldName10, fieldNumber: 410)
+      if _storage._fieldName10 != 0 {
+        try visitor.visitSingularInt32Field(value: _storage._fieldName10, fieldNumber: 410)
       }
-      if storage._fieldName11 != 0 {
-        try visitor.visitSingularInt32Field(value: storage._fieldName11, fieldNumber: 411)
+      if _storage._fieldName11 != 0 {
+        try visitor.visitSingularInt32Field(value: _storage._fieldName11, fieldNumber: 411)
       }
-      if storage._fieldName12 != 0 {
-        try visitor.visitSingularInt32Field(value: storage._fieldName12, fieldNumber: 412)
+      if _storage._fieldName12 != 0 {
+        try visitor.visitSingularInt32Field(value: _storage._fieldName12, fieldNumber: 412)
       }
-      if storage.__FieldName13 != 0 {
-        try visitor.visitSingularInt32Field(value: storage.__FieldName13, fieldNumber: 413)
+      if _storage.__FieldName13 != 0 {
+        try visitor.visitSingularInt32Field(value: _storage.__FieldName13, fieldNumber: 413)
       }
-      if storage.__FieldName14 != 0 {
-        try visitor.visitSingularInt32Field(value: storage.__FieldName14, fieldNumber: 414)
+      if _storage.__FieldName14 != 0 {
+        try visitor.visitSingularInt32Field(value: _storage.__FieldName14, fieldNumber: 414)
       }
-      if storage._field_Name15 != 0 {
-        try visitor.visitSingularInt32Field(value: storage._field_Name15, fieldNumber: 415)
+      if _storage._field_Name15 != 0 {
+        try visitor.visitSingularInt32Field(value: _storage._field_Name15, fieldNumber: 415)
       }
-      if storage._field_Name16 != 0 {
-        try visitor.visitSingularInt32Field(value: storage._field_Name16, fieldNumber: 416)
+      if _storage._field_Name16 != 0 {
+        try visitor.visitSingularInt32Field(value: _storage._field_Name16, fieldNumber: 416)
       }
-      if storage._fieldName17__ != 0 {
-        try visitor.visitSingularInt32Field(value: storage._fieldName17__, fieldNumber: 417)
+      if _storage._fieldName17__ != 0 {
+        try visitor.visitSingularInt32Field(value: _storage._fieldName17__, fieldNumber: 417)
       }
-      if storage._fieldName18__ != 0 {
-        try visitor.visitSingularInt32Field(value: storage._fieldName18__, fieldNumber: 418)
+      if _storage._fieldName18__ != 0 {
+        try visitor.visitSingularInt32Field(value: _storage._fieldName18__, fieldNumber: 418)
       }
     }
   }
 
   func _protoc_generated_isEqualTo(other: ProtobufTestMessages_Proto3_TestAllTypes) -> Bool {
-    return _storage === other._storage || _storage.isEqualTo(other: other._storage)
-  }
-
-  private mutating func _uniqueStorage() -> _StorageClass {
-    if !isKnownUniquelyReferenced(&_storage) {
-      _storage = _storage.copy()
+    return withExtendedLifetime((_storage, other._storage)) { (_storage, other_storage) in
```

Comment:
Actually, there's still a possible improvement: if we know the storage identities are the same, we don't have to check the individual fields in them. So while we can't short circuit the *whole thing*, we could at least wrap the checks of the `_storage` fields in a conditional that only runs if `_storage !== other_storage`.

The logic will get a little tricker when we start pulling subsets of fields out of storage, though, because we'll have to keep those separate instead of interleaving them.